roles/ceph-client/tasks/create_users_keys.yml

```diff
@@ -22,14 +24,19 @@
   failed_when: false
   with_items: "{{ keys }}"
   register: keys_exist
+  when:
+    - copy_admin_key
 
 - name: add key(s) to ceph
   command: "ceph --cluster {{ cluster }} auth import -i /etc/ceph/{{ cluster }}.{{ item.0.name }}.keyring"
   changed_when: false
+  ignore_errors: true
```

Comment:
I think we can remove this now right?

Reply:
If we remove ignore_errors:true, then item1.rc [1] will not be set because the previous task didn't run and then, according to my testing, the play will fail. If it's important to fail the play when the "add key(s) to ceph" task fails and not have ignore_errors:true, then one alternative is to have keys_exist.results set when not copy_admin_key. Should I add a task for that between "check if key(s) already exist(s)" and "add key(s) to ceph"? 

[1] https://github.com/fultonj/ceph-ansible/blob/b996c1f34e7c7e871b87e2295402ed667798e82b/roles/ceph-client/tasks/create_users_keys.yml#L39

Reply:
You need to do the following in your `when` condition:

```
- not item.1.get("skipped")
```

Finally change `"{{ keys_exist.results }}"` to `"{{ keys_exist.results | default([])}}"`

Please try, thanks!